src/main/java/sirius/pasta/noodle/SharedConstantPool.java

```diff
@@ -78,4 +77,13 @@ public synchronized int getIndex(Object constant) {
             return -1;
         }
     }
+
+    /**
+     * Returns a copy of the list of shared constants.
+     *
+     * @return a copy of all shared constants. Note that this list cannot be mutated.
+     */
+    public synchronized List<Object> getSharedConstants() {
+        return Collections.unmodifiableList(new ArrayList<>(sharedConstants));
```

Comment:
Is this double wrapping necessary? Wouldnt `Collections.unmodifiableList(sharedConstants)` work, too?

Reply:
Dachte erst ja, weil wir oben was adden. Aber sind ja beide Methoden synchronized

Reply:
The underlying list is a plain array list which can be modified concurrently. The unmodifyableList delegates all call to its given list, so we need to create a copy while being within the sync block...